### PR TITLE
Stop flaky Safaricom E2E

### PR DESCRIPTION
**Pull request by nielsVoogt**
[AB#44134](https://dev.azure.com/redcrossnl/7d17f2a9-3f18-40f1-8c29-98568fbca7e4/_workitems/edit/44134) 

## Describe your changes

Fixes flaky Safaricom - and possibly other payment related FSP tests - by making the waiting for payment completion more robust.  Also a minor update of README.MD with a fancy trick :)

## Checklist before requesting a code review

- [x] I have performed a self-review of my code
- [x] I have addressed all Copilot comments
- [x] The changes do not touch the UI/UX
- [x] I have edited tests for my changes
- [x] I have made sure that all automated checks pass before requesting a review
- [x] I do not need any deviation from [our PR guidelines](https://github.com/global-121/121-platform/blob/main/docs/CONTRIBUTING.md#pull-request-guidelines)

## Portal preview-deployment

https://happy-rock-0411d2003-8701.westeurope.3.azurestaticapps.net

### CODE REVIEW

File: e2e/portal/pages/PaymentPage.ts [hunk 1]
```diff
@@ -33,6 +34,9 @@ class PaymentPage extends BasePage {
     this.importReconciliationDataButton = this.page.getByRole('button', {
       name: 'Import reconciliation data',
     });
+    this.succesfullyTransferredChip = this.page.getByTestId(
+      'metric-tile-chip-successfully-transferred',
+    );
```
[the remark this answers is not in the document]
Comment (nielsVoogt) [reply e2e/portal/pages/PaymentPage.ts]: Picked up

File: e2e/portal/pages/PaymentPage.ts [hunk 2]
```diff
@@ -154,6 +160,24 @@ class PaymentPage extends BasePage {
...
+    /*
+      The method utilized above isn't that robust.
+      Because there is a inbetween state where the approved chip is visible,
+      but the payment is still 'processing'. Since we can't use the <canvas />
+      element to check the graph status, we need to check the amount that is
+      displayed in the payment card. When this chip shows the correct amount
+      we can safely navigate away and assert the payment card in the overview
+    */
+
+    if (expectedAmount) {
```
[the remark this answers is not in the document]
Comment (nielsVoogt) [reply e2e/portal/pages/PaymentPage.ts]: Picked up

File: e2e/portal/pages/PaymentPage.ts [hunk 3]
```diff
@@ -33,6 +34,10 @@ class PaymentPage extends BasePage {
     this.importReconciliationDataButton = this.page.getByRole('button', {
       name: 'Import reconciliation data',
     });
+    this.succesfullyTransferredAmountChip = this.page
+      .locator('app-metric-tile', { hasText: 'Total amount' })
+      .getByTestId('metric-tile-chip');
```
Comment (elwinschmitz) [inline e2e/portal/pages/PaymentPage.ts]: When using the 'Test ID', it wouldn't be necessary anymore to use another (implementation-specific) Locator, right?

(Would it be possible to add the `hasText`-filter to the `getByTestId`-method?)

In general, I think it is nice to have as much 'implementation-agnostic'-tests as possible (and thus using `getByRole`-like locators); But in practice, there are already quite some tests using these specific selectors/locators.... So maybe not necessary to 'solve everything right here, right now'...

File: e2e/README.md [hunk 4]
```diff
@@ -84,6 +84,16 @@ npm run test:find-flaky <substringMatchingFilename>
...
 
+Another great tip for finding difficult edge cases is [CPU throttling](https://charpeni.com/blog/how-to-easily-reproduce-a-flaky-test-in-playwright)
+
+```ts
+test.beforeEach(async ({page }) => {
+  const context = page.context();
+  const cdpSession = await context.newCDPSession(page);
+  await cdpSession.send('Emulation.setCPUThrottlingRate', { rate: 6 });
+)
+```
```
Comment (elwinschmitz) [inline e2e/README.md:95]: 🎉

File: e2e/portal/pages/PaymentPage.ts [hunk 5]
```diff
@@ -154,6 +161,26 @@ class PaymentPage extends BasePage {
 
     await inProgressChip.waitFor({ state: 'hidden' });
     await approvedChip.waitFor({ state: 'visible' });
+
+    /*
+      The method utilized above isn't that robust.
+      Because there is a inbetween state where the approved chip is visible,
```
Comment (elwinschmitz) [inline e2e/portal/pages/PaymentPage.ts]:
```suggestion
      Because there is an in-between state where the "Approved"-chip is visible,
```

File: e2e/portal/pages/PaymentPage.ts [hunk 6]
```diff
@@ -154,6 +161,26 @@ class PaymentPage extends BasePage {
...
+      correct amount we can safely navigate away and assert the payment card in
+      the overview.
+    */
+
+    if (expectedAmount !== undefined) {
+      await expect
+        .poll(async () => {
+          const chipText =
+            await this.succesfullyTransferredAmountChip.textContent();
+          return chipText?.replace(/,/g, '');
```
Comment (elwinschmitz) [inline e2e/portal/pages/PaymentPage.ts:179]: This is to get rid of any "currency formatting", right?

**Review by elwinschmitz — DISMISSED** [review]
(All comments are mainly optional. ;) )

File: e2e/portal/pages/PaymentPage.ts (hunk 6 above)
Comment (nielsVoogt) [reply e2e/portal/pages/PaymentPage.ts:179]: Correct

File: e2e/portal/pages/PaymentPage.ts (hunk 3 above)
Comment (nielsVoogt) [reply e2e/portal/pages/PaymentPage.ts]: Yes, you are right. We don't need the specificity here because other `tiles` that use this `chip` as well are  in another view.

File: e2e/portal/pages/PaymentPage.ts (hunk 5 above)
Comment (nielsVoogt) [reply e2e/portal/pages/PaymentPage.ts]: Picked up

**Review by elwinschmitz — APPROVED** [review]
